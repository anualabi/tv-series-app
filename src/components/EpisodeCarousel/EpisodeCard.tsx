import React from 'react'
import Image from 'next/image';

import { EpisodeProps } from '@/types';

export type EpisodeCardProps = Pick<EpisodeProps, 'Poster' | 'Episode' | 'Title' | 'Plot'>

const EpisodeCard: React.FC<EpisodeCardProps> = ({ Poster, Episode, Title, Plot }) => {
  return (
    <div className="overflow-hidden w-56">
      <div className="relative w-[201px] h-[134px]">
        <Image
          src={Poster}
          alt={`${Title} image`}
          style={{objectFit: "cover"}}	
          className="w-full h-full"
          width={201}
          height={134}
        />
        <div className="absolute top-0 left-0 bg-white w-[30px] h-[30px] flex items-center justify-center">
          <p className="text-sm font-bold text-black">{Episode}</p>
        </div>
      </div>
      <div className="py-4 bg-opacity-50 relative w-[201px] h-[138px]">
        <h2 className="text-sm md:text-base  text-white font-bold mb-2 line-clamp-1">{Title}</h2>
        <p className='text-xs md:text-sm text-white font-light line-clamp-3'>{Plot}</p>
      </div>
    </div>
  )
}

export default EpisodeCard
