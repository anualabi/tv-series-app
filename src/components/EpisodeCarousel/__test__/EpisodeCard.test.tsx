import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import EpisodeCard, { EpisodeCardProps } from '../EpisodeCard'; 

const mockProps = {
    Poster: 'https://test-poster.jpg',
    Episode: '5',
    Title: 'Better-Like',
    Plot: 'This is a sample plot for testing purposes.'
};

const customRender = (props: EpisodeCardProps) => render(<EpisodeCard {...props} />);

describe('EpisodeCard', () => {
    test('should render snapshot', () => {
        const { asFragment } = customRender({...mockProps})
        
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the episode number correctly', () => {
        customRender({ ...mockProps })
        
        const episodeNumber = screen.getByText('5');

        expect(episodeNumber).toBeInTheDocument();
    });

    test('renders the title correctly', () => {
        customRender({ ...mockProps })

        const title = screen.getByText('Better-Like');

        expect(title).toBeInTheDocument();
    });

    test('renders the plot correctly', () => {
        customRender({ ...mockProps })
        
        const plot = screen.getByText(/This is a sample plot for testing purposes./i);

        expect(plot).toBeInTheDocument();
    });
});
