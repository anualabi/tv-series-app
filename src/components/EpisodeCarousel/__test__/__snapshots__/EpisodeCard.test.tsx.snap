// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeCard should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="overflow-hidden w-56"
  >
    <div
      class="relative w-[201px] h-[134px]"
    >
      <img
        alt="Better-Like image"
        class="w-full h-full"
        data-nimg="1"
        decoding="async"
        height="134"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Ftest-poster.jpg&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Ftest-poster.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Ftest-poster.jpg&w=640&q=75 2x"
        style="color: transparent; object-fit: cover;"
        width="201"
      />
      <div
        class="absolute top-0 left-0 bg-white w-[30px] h-[30px] flex items-center justify-center"
      >
        <p
          class="text-sm font-bold text-black"
        >
          5
        </p>
      </div>
    </div>
    <div
      class="py-4 bg-opacity-50 relative w-[201px] h-[138px]"
    >
      <h2
        class="text-sm md:text-base  text-white font-bold mb-2 line-clamp-1"
      >
        Better-Like
      </h2>
      <p
        class="text-xs md:text-sm text-white font-light line-clamp-3"
      >
        This is a sample plot for testing purposes.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
